lib/spack/spack/s3_handler.py

```diff
@@ -65,23 +65,7 @@ def s3_open(self, req):
         try:
             url, headers, stream = _s3_open(orig_url)
             return urllib_response.addinfourl(stream, headers, url)
-        except ClientError as err:
-            # if no such [KEY], but [KEY]/index.html exists,
-            # return that, instead.
-            if err.response['Error']['Code'] == 'NoSuchKey':
-                try:
-                    _, headers, stream = _s3_open(
-                        url_util.join(orig_url, 'index.html'))
-                    return urllib_response.addinfourl(
-                        stream, headers, orig_url)
-
-                except ClientError as err2:
-                    if err.response['Error']['Code'] == 'NoSuchKey':
-                        # raise original error
-                        raise urllib_error.URLError(err)
-
-                    raise urllib_error.URLError(err2)
-
+        except (ClientError, ValueError) as err:
```

Comment:
This is for my understanding of the code. Why do we intercept these exceptions and re-raise them as `URLError`?

Reply:
This needs to be done because this method is called as part of `urllib`.  Their API [dictates](https://docs.python.org/3/library/urllib.error.html?highlight=urllib#urllib.error.URLError) that any `urlopen` handler that does not successfully handle a request raises a `URLError`